relayer/pkg/chainlink/txm/nonce.go

```diff
@@ -64,60 +64,72 @@ func (nm *nonceManager) HealthReport() map[string]error {
 	return map[string]error{nm.Name(): nm.starter.Healthy()}
 }
 
-// Register is used because we cannot pre-fetch nonces. the pubkey is known before hand, but the account address is not known until a job is started and sends a tx
-func (nm *nonceManager) Register(ctx context.Context, addr *felt.Felt, publicKey *felt.Felt, chainId string, client NonceManagerClient) error {
+func (nm *nonceManager) Sync(ctx context.Context, address *felt.Felt, publicKey *felt.Felt, client NonceManagerClient) error {
 	nm.lock.Lock()
 	defer nm.lock.Unlock()
-	addressNonces, exists := nm.n[publicKey.String()]
-	if !exists {
-		nm.n[publicKey.String()] = map[string]*felt.Felt{}
+
+	if err := nm.validate(publicKey); err != nil {
+		return err
+	}
+
+	n, err := client.AccountNonce(ctx, address)
+	if err != nil {
+		return err
 	}
-	_, exists = addressNonces[chainId]
+
+	nm.n[publicKey.String()] = n
+
+	return nil
+}
+
+// Register is used because we cannot pre-fetch nonces. the pubkey is known before hand, but the account address is not known until a job is started and sends a tx
+func (nm *nonceManager) Register(ctx context.Context, addr *felt.Felt, publicKey *felt.Felt, client NonceManagerClient) error {
+	nm.lock.Lock()
+	defer nm.lock.Unlock()
+
+	_, exists := nm.n[publicKey.String()]
 	if !exists {
 		n, err := client.AccountNonce(ctx, addr)
 		if err != nil {
 			return err
 		}
-		nm.n[publicKey.String()][chainId] = n
+		nm.n[publicKey.String()] = n
 	}
 
 	return nil
 }
 
-func (nm *nonceManager) NextSequence(addr *felt.Felt, chainId string) (*felt.Felt, error) {
-	if err := nm.validate(addr, chainId); err != nil {
+func (nm *nonceManager) NextSequence(publicKey *felt.Felt) (*felt.Felt, error) {
+	nm.lock.RLock()
+	defer nm.lock.RUnlock()
+
+	if err := nm.validate(publicKey); err != nil {
 		return nil, err
 	}
 
-	nm.lock.RLock()
-	defer nm.lock.RUnlock()
-	return nm.n[addr.String()][chainId], nil
+	return nm.n[publicKey.String()], nil
```

Comment:
Minor nit, in both cases we validate which is a map lookup, then immediately fetch again. We could rename 'validate' into 'fetch' and have it return the value (we can ignore the value returned in 'Sync')